fem/complex_fem.cpp

```diff
@@ -781,59 +932,153 @@ ParSesquilinearForm::FormLinearSystem(const Array<int> &ess_tdof_list,
    Vector b_i(&(b.GetData())[vsize], vsize);
    b_i *= s;
 
-   OperatorHandle A_r, A_i;
-   Vector X_0, B_0;
+   int tvsize = pfes->GetTrueVSize();
 
-   b_0 = b_r;
-   pblfr->FormLinearSystem(ess_tdof_list, x_r, b_0, A_r, X_0, B_0, ci);
+   OperatorHandle A_r, A_i;
 
-   int tvsize = B_0.Size();
    X.SetSize(2 * tvsize);
    B.SetSize(2 * tvsize);
-   Vector X_r(X.GetData(), tvsize);
+
+   Vector X_0(tvsize), B_0(tvsize);
+   Vector X_r(X.GetData(),tvsize);
    Vector X_i(&(X.GetData())[tvsize], tvsize);
    Vector B_r(B.GetData(), tvsize);
    Vector B_i(&(B.GetData())[tvsize], tvsize);
-   X_r = X_0; B_r = B_0;
 
-   b_0 = 0.0;
-   pblfi->FormLinearSystem(ess_tdof_list, x_i, b_0, A_i, X_0, B_0, false);
-   B_r -= B_0;
+   if (RealInteg() && ImagInteg())
```

Comment:
The same elimination of redundant code in the serial class can be done here.

Reply:
See the comment in the serial case